datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -1958,6 +1964,90 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn row_group_predicate_between() -> Result<()> {
```

Comment:
Add some test case.  because found 
```
ID IN (46402206,
                         201143645,
                         1147370581,
....
                         242375670,
                         38453705)
```
Before  found `MAX_LIST_VALUE_SIZE_REWRITE  ` I though this filter not push down is wrong 😭